tools/depcheck/pkg/graph/flags.go

```diff
@@ -0,0 +1,308 @@
+package graph
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"path"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/golang/glog"
+
+	"github.com/gonum/graph/concrete"
+)
+
+// GraphOptions contains values necessary to create a dependency graph.
+type GraphOptions struct {
+	// Entrypoints are paths used to begin scanning for go packages
+	Entrypoints []string
+	// RepoImportPath is the "go-import path" of the repository being traversed
+	RepoImportPath string
+
+	// Packages is the caculated list of traversed go packages to use to create the graph
+	Packages *PackageList
+
+	// Roots is a list of go-import paths containing the total set of traversed
+	// packages, from the given set of Entrypoints.
+	Roots []string
```

Comment:
What is the difference between entrypoints and roots?

Reply:
Entrypoints are the unparsed paths provided via `--entry` by the user.
Roots are calculated before the graph is build based on a traversal of all packages reachable from each Entrypoint

Reply:
> Entrypoints are the unparsed paths provided via --entry by the user.
> Roots are calculated before the graph is built based on a traversal of all packages reachable from each Entrypoint

Why does `options` need the unprocessed user input.  That defies my expectations.

Reply:
> What is the difference between entrypoints and roots?

resolved